### PR TITLE
Recipes belong to users

### PR DESCRIPTION
**Pull request by moscc**
### What we did?
- Added relation between recipe and user models
- Added user_id attribute to recipe model
- Display author email in show and index views
- Modified existing tests to fit the new model modification 

### What we have learned?
- That we have to modify the old tests and run them before pushing, to make sure they all pass.

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -6,5 +6,115 @@
...
+  # name: 'George',
+  email: 'george@example.com',
+  password: '123456'
+)
+
+# Category.create([{name: 'Appetizers & Snacks'},
+#                   {name: 'Breakfast & Brunch'},
+#                   {name: 'Desserts'},
+#                   {name: 'Dinner'},
+#                   {name: 'Drinks'}])
```
Comment (tochman) [inline db/seeds.rb]: Please, avoid comments

File: db/seeds.rb [hunk 2]
```diff
@@ -6,5 +6,115 @@
...
-Recipe.create(title: "Meatball-Stew", description: "will make you sick",
-              ingredients: "meat, onion, stuff", directions: "stir it, mix it, eat it")
+User.create(
+  # name: 'Nader',
+  email: 'nader@example.com',
+  password: '123456'
+)
+
+User.create(
+  # name: 'George',
```
Comment (tochman) [inline db/seeds.rb]: Please avoid comments

File: features/recipes_belong_to_users.feature [hunk 3]
```diff
@@ -0,0 +1,21 @@
+Feature:
+  As a user
+  In order to protect recipes from editing by others
+  My recipes should belong to me
```
Comment (tochman) [inline features/recipes_belong_to_users.feature]: Please stick to the Connextra format

File: features/recipes_belong_to_users.feature [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+  Background:
+    Given We have the following user:
+      | email           | password    |
+      | maran@test.com  | 12345678    |
+    And We have the following recipes:
+      | title       	| description                 | ingredients        | directions              |
+      | Meatball-Stew | This will make you sick     | Meat, onion, stuff | Stir it, mix it, eat it |
```
Comment (tochman) [inline features/recipes_belong_to_users.feature]: If you don't test for ingredients and/or directions, please let the Factory do it's job

File: features/recipes_belong_to_users.feature [hunk 5]
```diff
@@ -0,0 +1,21 @@
+Feature:
+  As a user
+  In order to protect recipes from editing by others
+  My recipes should belong to me
+
+  Background:
+    Given We have the following user:
+      | email           | password    |
+      | maran@test.com  | 12345678    |
+    And We have the following recipes:
```
Comment (tochman) [inline features/recipes_belong_to_users.feature]: Please insert a line break between steps

File: features/recipes_belong_to_users.feature [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+      | title       	| description                 | ingredients        | directions              |
+      | Meatball-Stew | This will make you sick     | Meat, onion, stuff | Stir it, mix it, eat it |
+
+  Scenario: Visitor can see authors email on index page
+    Given I visit the site
+    Then I should see "maran@test.com"
```
Comment (tochman) [inline features/recipes_belong_to_users.feature]: Perhaps look for "Posted by: maran@test.com"?
Also, I would add more than 1 recipe in the background step and check for the creator on each div. I can show you how it's done

File: features/show_recipe_in_a_view.feature [hunk 7]
```diff
@@ -4,11 +4,13 @@ Feature: Show recipe in a view (User views recipe in detail)
   I should be able to view a single recipe with all details
 
   Background:
-    Given We have the following list:
+    Given We have the following user:
+      | email           | password    |
+      | maran@test.com  | 12345678    |
+    And We have the following recipes:
```
Comment (tochman) [inline features/show_recipe_in_a_view.feature]: Please insert line break between steps

File: features/step_definitions/basic_steps.rb [hunk 8]
```diff
@@ -20,10 +20,9 @@
   end
 end
 
-Given(/^We have the following (?:recipes|list):$/) do |table|
+Given('We have the following recipes:') do |table|
   table.hashes.each do |recipe|
+    recipe["user_id"] = User.last.id
```
Comment (tochman) [inline features/step_definitions/basic_steps.rb]: Please use a symbol for the key. Do you really want the same user to be assigned to the recipe like this? If yes, let the Factory do its job

File: spec/factories/recipe.rb [hunk 9]
```diff
@@ -1,8 +1,9 @@
-FactoryBot.define do
+ FactoryBot.define do
   factory :recipe do
     title 'MyRecipe'
     description 'A fake recipe to try cook something'
     ingredients 'Eggs, flour, tomato, meat, other stuff'
     directions 'Put stuff together and mix them, then bake them'
+    user User.create(email: 'maran@dream.com', password: '123456')
```
Comment (tochman) [inline spec/factories/recipe.rb]: Please research how you can use associations in factories. This is not the way to go.

File: spec/models/recipe_spec.rb [hunk 10]
```diff
@@ -8,6 +8,7 @@
     it { is_expected.to have_db_column :description}
     it { is_expected.to have_db_column :ingredients}
     it { is_expected.to have_db_column :directions}
+    it { is_expected.to have_db_column :user_id}
```
Comment (tochman) [inline spec/models/recipe_spec.rb:11]: Please add specs for associations using ShouldaMatcher

**Review by tochman — CHANGES_REQUESTED** [review]

File: features/recipes_belong_to_users.feature (hunk 6 above)
Comment (alghazu) [reply features/recipes_belong_to_users.feature]: Done

File: features/step_definitions/basic_steps.rb [hunk 11]
```diff
@@ -15,15 +15,14 @@
 end
 
 Given('We have the following user:') do |table|
-  table.hashes.each do |_field|
-    create(:user)
+  table.hashes.each do |field|
+    FactoryBot.create(:user, field)
```
Comment (tochman) [inline features/step_definitions/basic_steps.rb]: Why do you revert to include this. The configuration to include FactoryBot methods was done in another PR

Comment (alghazu) [reply features/step_definitions/basic_steps.rb]: I didn't really get your comment.
And also no idea what does underscore in `|_feild|` does. 
It was not creating the required users when debugging, so changed it, believing this is the simplest way to create users and recipes using FactoryBot.

Comment (tochman) [reply features/step_definitions/basic_steps.rb]: The question is why you changed `create(:user)` to `FactoryBot.create(:user, field)` and not `.create(:user, field)` (including the module name `FactoryBot`

Comment (alghazu) [reply features/step_definitions/basic_steps.rb]: Got it. Done.

File: spec/models/recipe_spec.rb (hunk 10 above)
Comment (alghazu) [reply spec/models/recipe_spec.rb:11]: Done
